src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -342,7 +347,11 @@ protected function instantiateObject(array &$data, string $class, array &$contex
 
         $constructor = $this->getConstructor($data, $class, $context, $reflectionClass, $allowedAttributes);
         if ($constructor) {
-            if (true !== $constructor->isPublic()) {
+            $createInstanceWithoutConstructor = (
+                \array_key_exists(self::CREATE_INSTANCE_WITHOUT_CONSTRUCTOR, $context)
+                && (true === $context[self::CREATE_INSTANCE_WITHOUT_CONSTRUCTOR])
+            );
+            if (true !== $constructor->isPublic() || $createInstanceWithoutConstructor) {
```

Comment:
this should written as 
`if (!$constructor->isPublic() || ($context[self::CREATE_INSTANCE_WITHOUT_CONSTRUCTOR] ?? $this->defaultContext[self::CREATE_INSTANCE_WITHOUT_CONSTRUCTOR] ?? false)) {`